// panasonic-comfort-cloud-client/dist/src/domain/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Power;
(function (Power) {
    Power[Power["Off"] = 0] = "Off";
    Power[Power["On"] = 1] = "On";
})(Power = exports.Power || (exports.Power = {}));
var OperationMode;
(function (OperationMode) {
    OperationMode[OperationMode["Auto"] = 0] = "Auto";
    OperationMode[OperationMode["Dry"] = 1] = "Dry";
    OperationMode[OperationMode["Cool"] = 2] = "Cool";
    OperationMode[OperationMode["Heat"] = 3] = "Heat";
    OperationMode[OperationMode["Fan"] = 4] = "Fan";
})(OperationMode = exports.OperationMode || (exports.OperationMode = {}));
var AirSwingUD;
(function (AirSwingUD) {
    AirSwingUD[AirSwingUD["Up"] = 0] = "Up";
    AirSwingUD[AirSwingUD["UpMid"] = 3] = "UpMid";
    AirSwingUD[AirSwingUD["Mid"] = 2] = "Mid";
    AirSwingUD[AirSwingUD["DownMid"] = 4] = "DownMid";
    AirSwingUD[AirSwingUD["Down"] = 1] = "Down";
})(AirSwingUD = exports.AirSwingUD || (exports.AirSwingUD = {}));
var AirSwingLR;
(function (AirSwingLR) {
    AirSwingLR[AirSwingLR["Left"] = 0] = "Left";
    AirSwingLR[AirSwingLR["LeftMid"] = 4] = "LeftMid";
    AirSwingLR[AirSwingLR["Mid"] = 2] = "Mid";
    AirSwingLR[AirSwingLR["RightMid"] = 3] = "RightMid";
    AirSwingLR[AirSwingLR["Right"] = 1] = "Right";
})(AirSwingLR = exports.AirSwingLR || (exports.AirSwingLR = {}));
var EcoMode;
(function (EcoMode) {
    EcoMode[EcoMode["Auto"] = 0] = "Auto";
    EcoMode[EcoMode["Powerful"] = 1] = "Powerful";
    EcoMode[EcoMode["Quiet"] = 2] = "Quiet";
})(EcoMode = exports.EcoMode || (exports.EcoMode = {}));
var FanSpeed;
(function (FanSpeed) {
    FanSpeed[FanSpeed["Auto"] = 0] = "Auto";
    FanSpeed[FanSpeed["Low"] = 1] = "Low";
    FanSpeed[FanSpeed["LowMid"] = 2] = "LowMid";
    FanSpeed[FanSpeed["Mid"] = 3] = "Mid";
    FanSpeed[FanSpeed["HighMid"] = 4] = "HighMid";
    FanSpeed[FanSpeed["High"] = 5] = "High";
})(FanSpeed = exports.FanSpeed || (exports.FanSpeed = {}));
var dataMode;
(function (dataMode) {
    dataMode[dataMode["Day"] = 0] = "Day";
    dataMode[dataMode["Week"] = 1] = "Week";
    dataMode[dataMode["Month"] = 2] = "Month";
    dataMode[dataMode["Year"] = 4] = "Year";
})(dataMode = exports.dataMode || (exports.dataMode = {}));
var FanAutoMode;
(function (FanAutoMode) {
    FanAutoMode[FanAutoMode["Disabled"] = 1] = "Disabled";
    FanAutoMode[FanAutoMode["AirSwingAuto"] = 0] = "AirSwingAuto";
    FanAutoMode[FanAutoMode["AirSwingLR"] = 3] = "AirSwingLR";
    FanAutoMode[FanAutoMode["AirSwingUD"] = 2] = "AirSwingUD";
})(FanAutoMode = exports.FanAutoMode || (exports.FanAutoMode = {}));
